pkg/oomstore/cache.go

```diff
@@ -28,9 +28,10 @@ type GroupBuffer struct {
 }
 
 type StreamPushProcessor struct {
-	capacity  int
-	minPeriod time.Duration
-	quit      chan bool
+	capacity          int
+	minPeriod         time.Duration
+	quit              chan bool
+	waitProcessorQuit chan struct{}
```

Comment:
Are both `quit` and `waitProcessorQuit` required ?

Reply:
Or `notifyQuit` and `waitQuit`? BTW can we use the same type (choose one of `bool` or `struct{}`) ?

Reply:
sound good! I will use `struct{}` since it doesn't need memory.